components/frontend/src/report/ReportsOverview.js

```diff
@@ -113,35 +167,54 @@ export function ReportsOverview({ dates, hiddenTags, measurements, reports, open
                 hiddenTags={hiddenTags}
                 layout={reports_overview.layout ?? []}
                 measurements={reversedMeasurements}
-                open_report={open_report}
+                onClickTag={(tag) => {
+                    const tagsToToggle = hiddenTags?.length > 0 ? hiddenTags : getReportsTags(reports)
```

Comment:
I don't really understand what is happening here from the naming. Perhaps add a comment?